pyppeteer/target.py

```diff
@@ -58,19 +73,28 @@ def _closedCallback(self) -> None:
         If the target is not of type "page" or "background_page", return
         ``None``.
         """
-        if (self._targetInfo['type'] in ['page', 'background_page'] and
-                self._page is None):
-            client = await self._sessionFactory()
-            new_page = await Page.create(
-                client, self,
+        if self._page:
+            return self._page
+        if self._targetInfo['type'] in ['page', 'background_page']:
+            session = await self._sessionFactory()
+            self._page = await Page.create(
+                session,
+                self,
                 self._ignoreHTTPSErrors,
                 self._defaultViewport,
                 self._screenshotTaskQueue,
             )
-            self._page = new_page
-            return new_page
         return self._page
 
+    async def worker(self):
+        _type = self._targetInfo['type']
+        if _type not in ['service_worker', 'shared_worker']:
+            return
+        if not self._workerPromise:
+            session = self._sessionFactory()
```

Comment:
This should be awaited, based on previous usage anyway.
```suggestion
            session = await self._sessionFactory()
```